File: python/fastapi/src/app/modules/branchs/repository.py
from app.database import db

# Entity
from .entity import BranchEntity

class BranchsRepository:
    def __init__(self):
        self._db = db

    def find_all(self) -> list[BranchEntity]:
        with self._db.cursor() as cursor:
            cursor.execute('SELECT * FROM "Agencia"')
            columns = [col[0] for col in cursor.description]
            rows = cursor.fetchall()
            return [dict(zip(columns, row)) for row in rows]
    
    def find_by_id(self, id: str) -> BranchEntity:
        with self._db.cursor() as cursor:
            cursor.execute('SELECT * FROM "Agencia" a WHERE a.id = %s LIMIT 1', [id])
            columns = [col[0] for col in cursor.description]
            row = cursor.fetchone()
            return dict(zip(columns, row)) if row else None

File: go/gin/internal/modules/branch/repository.go
package branch

import (
	"context"
	"log" // Adicionado para log de erros
	"tcc/internal/database"
	"tcc/internal/modules/branch/entity"
	"time"
)

type BranchRepository struct {
}

func (r *BranchRepository) FindById(id string) (*entity.BranchEntity, error) {
	query := `SELECT * FROM "Agencia" a WHERE a.id = $1 LIMIT 1`
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
	defer cancel()
	row := database.Conn.QueryRow(ctx, query, id)

	var branch entity.BranchEntity
	err := row.Scan(&branch.ID, &branch.Nome, &branch.Telefone, &branch.Numero, &branch.DataDeCriacao, &branch.DataDeAtualizacao)
	if err != nil {
		log.Printf("Erro ao buscar agência por ID: %v", err) // Log de erro
		return nil, err
	}

	return &branch, nil
}

func (r *BranchRepository) FindAll() ([]*entity.BranchEntity, error) {
	query := `SELECT * FROM "Agencia"`

	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
	defer cancel()
	rows, err := database.Conn.Query(ctx, query)
	if err != nil {
		log.Printf("Erro ao buscar agências: %v", err) // Log de erro
		return nil, err
	}
	defer rows.Close()

	var branches []*entity.BranchEntity
	for rows.Next() {
		var branch entity.BranchEntity
		err := rows.Scan(&branch.ID, &branch.Nome, &branch.Telefone, &branch.Numero, &branch.DataDeCriacao, &branch.DataDeAtualizacao)
		if err != nil {
			log.Printf("Erro ao escanear agências: %v", err) // Log de erro
			return nil, err
		}
		branches = append(branches, &branch)
	}

	if err = rows.Err(); err != nil {
		log.Printf("Erro ao iterar sobre as linhas: %v", err) // Log de erro
		return nil, err
	}

	return branches, nil
}